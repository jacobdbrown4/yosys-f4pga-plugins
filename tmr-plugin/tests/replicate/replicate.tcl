yosys -import 
# if { [info procs ls] == {} } { plugin -i tmr }
read_verilog $::env(DESIGN_TOP).v
hierarchy -top toggle

# yosys synth_xilinx

plugin -i tmr
yosys -import

# yosys replicate

yosys cd toggle
yosys ls

set fp [open [test_output_path "replicate.txt"] "w"]

puts "hi"
puts $fp "hi"
puts $fp [replicate]

# puts $fp [yosys ls > fp]
# yosys tee -o ~/yosys_plugins/yosys-f4pga-plugins/testing.txt ls
# puts [yosys ls]
puts $fp [replicate]

close $fp