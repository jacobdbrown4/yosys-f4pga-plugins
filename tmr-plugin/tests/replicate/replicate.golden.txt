14 wires:
  _00__TMR_0
  _00__TMR_1
  _00__TMR_2
  _01_
  _02__TMR_0
  _02__TMR_1
  _02__TMR_2
  _03_
  _04__TMR_0
  _04__TMR_1
  _04__TMR_2
  clk
  out
  reset

12 cells:
  _05__TMR_0
  _05__TMR_1
  _05__TMR_2
  _06__TMR_0
  _06__TMR_1
  _06__TMR_2
  _07__TMR_0
  _07__TMR_1
  _07__TMR_2
  _08_
  _09_
  _10_
